src/main/java/org/takes/http/BkBasic.java

```diff
@@ -70,19 +76,43 @@ public BkBasic(final Take tks) {
         this.take = tks;
     }
 
+    // @todo #519:30min Need to support Keep-Alive header here and take
+    //  into account specified timeout and max requests parameters.
+    //  See http://tools.ietf.org/id/draft-thomson-hybi-http-timeout-01.html
+    //  for details. Also need to remove input.available <= 0 condition
+    //  - this is done in order to prevent failure of some tests,
+    //  should be replaced with proper Keep-Alive header in tests
     @Override
+    @SuppressWarnings({"PMD.EmptyCatchBlock",
+        "PMD.AvoidInstantiatingObjectsInLoops"})
     public void accept(final Socket socket) throws IOException {
+        socket.setSoTimeout(Tv.THOUSAND);
```

Comment:
@yegor256 I think it's reasonable temporary value(1 second) at this stage. Once we have Keep-Alive header support with right `max` and `timeout` we can review this default value. Having too big value will lead to very long build (because each test will be waiting given number of seconds)